sky/backends/backend_utils.py

```diff
@@ -1605,13 +1608,18 @@ def _query_status_gcp(
     # GCP does not clean up preempted TPU VMs. We remove it ourselves.
     # TODO(wei-lin): handle multi-node cases.
     if use_tpu_vm and len(status_list) == 0:
-        backend = backends.CloudVmRayBackend()
-        handle = global_user_state.get_handle_from_cluster_name(cluster)
-        backend.teardown_no_lock(handle,
-                                 terminate=True,
-                                 purge=False,
-                                 post_teardown_cleanup=False)
-
+        logger.debug(f'Terminating preempted TPU VM cluster {cluster}')
+        # Do not use CloudVMRayBackend.teardown_no_lock, as that will
+        # cause inifinite recursion by calling cluster status refresh
+        # again.
+        terminate_cmd = tpu_utils.terminate_tpu_vm_cluster_cmd(cluster, zone)
+        returncode, stdout, stderr = log_lib.run_with_log(terminate_cmd,
+                                                          os.devnull,
+                                                          shell=True,
+                                                          stream_logs=False,
+                                                          require_outputs=True)
+        subprocess_utils.handle_returncode(returncode, terminate_cmd,
```

Comment:
One downside is this can exit the refresh code path. Calling `backend.teardown_no_lock` may hint about --purge.

Wdyt about the alternative of still calling `backend.teardown_no_lock` but use something like `do_status_refresh=False`?

Reply:
Good point! Fixed. Thanks!

Reply:
Actually, how do you feel about the current `refresh_cluster_status` argument for the `backend.teardown_no_lock` function? The TPU VM case only uses it in the cluster status refresh, which makes the API not look very clean to me.